### DISCUSSION TITLE
Unable to get simplest example working

### DISCUSSION THREAD

**Discussion by dudeNumber4** [g4s8/tuples#38] [Q&A] [answered]
Trying to use Pair.  Using jshell 17, passing your jar in --class-path.  Code follows.  What am I missing?

```
import wtf.g4s8.tuples.Pair;
public class B
{
    public static void b()
    {
        Pair<Integer, String> p = Pair.of(1, "you gotta have a pair");
        p.apply((i, s) -> System.out.println(s));
    }
}
```
Error:
incompatible types: inference variable R has incompatible bounds
    lower bounds: java.lang.Object
    lower bounds: void

**Comment by g4s8** [comment, marked as answer]
@dudeNumber4 you are looking for `accept()` method: https://github.com/g4s8/tuples/blob/master/src/main/java/wtf/g4s8/tuples/Pair.java#L24 

`apply()` expects a bi-function with a result: `BiFunction<? super T, ? super U, ? extends R>`, it can't accept `void` consumer like `println`, but `accept` takes `BiConsumer<? super T, ? super U>` argument.

This example works fine:
```java
var p = Pair.of(1, "you gotta have a pair");
p.accept((i, s) -> System.out.println(s));
```
